lib/fp.js

```diff
@@ -54,13 +40,19 @@ const fmap = (fn1, f) => {
 //   funcA - <Function>
 const ap = (fn, funcA) => concat(funcA, (f, callback) => fmap(fn, f)(callback));
 
-asyncChainMethods = { fmap, ap, concat };
+function Functor() {}
+
+const FP_FUNCTION_METHODS = [concat, fmap, ap];
+for (const method of FP_FUNCTION_METHODS) {
+  Functor.prototype[method.name] = function(...args) {
```

Comment:
I'd like to rename `Functor.prototype.fmap` to `Functor.prototype.map` because it is common for javascript. But `metasync.fmap(f, g)` is a good name and we don't need to rename it to `map`.

Reply:
Excellent point. Thanks